packages/core/http/core-http-router-server-internal/src/router.ts

```diff
@@ -202,6 +205,14 @@ export class Router<Context extends RequestHandlerContextBase = RequestHandlerCo
       return hapiResponseAdapter.toInternalError();
     }
   }
+
+  private versionedRouter: undefined | VersionedRouter<Context> = undefined;
+  public get versioned(): VersionedRouter<Context> {
+    if (this.versionedRouter === undefined) {
+      this.versionedRouter = CoreVersionedRouter.from({ router: this });
+    }
+    return this.versionedRouter;
+  }
```

Comment:
This is the implementation for exposing the versioned router

Reply:
Based on recent discussions around architectural complexity specifically related to core's packages footprint, perhaps we should move the versioned router into the main `core-http-router-server-internal` package.